src/Test.sol

```diff
@@ -448,8 +449,471 @@ abstract contract Test is DSTest, Script {
+        bytes memory parsedDeployData = vm.parseJson(deployData, ".receipts");
+        RawReceipt[] memory rawReceipts = abi.decode(parsedDeployData, (RawReceipt[]));
+        return rawToConvertedReceipts(rawReceipts);
+    }
+
+    function readReceipt(string memory path, uint index)
+        internal
+        returns (Receipt memory)
+    {
+        string memory deployData = vm.readFile(path);
+        string memory key = string(abi.encodePacked(".receipts[",vm.toString(index), "]"));
+        bytes memory parsedDeployData = vm.parseJson(deployData, key);
+        RawReceipt memory rawReceipt = abi.decode(parsedDeployData, (RawReceipt));
+        return rawToConvertedReceipt(rawReceipt);
+    }
+
+    function rawToConvertedReceipts(RawReceipt[] memory rawReceipts)
+        internal
+        returns(Receipt[] memory)
+    {
+        Receipt[] memory receipts = new Receipt[](rawReceipts.length);
+        for(uint i;i<rawReceipts.length;i++){
+            receipts[i] = rawToConvertedReceipt(rawReceipts[i]);
+        }
+        return receipts;
+    }
+
+    function rawToConvertedReceipt(RawReceipt memory rawReceipt)
+        internal
+        returns(Receipt memory)
+    {
+        Receipt memory receipt;
+        receipt.blockHash = rawReceipt.blockHash;
+        receipt.to = rawReceipt.to;
+        receipt.from = rawReceipt.from;
+        receipt.contractAddress = rawReceipt.contractAddress;
+        receipt.effectiveGasPrice = bytesToUint(rawReceipt.effectiveGasPrice);
+        receipt.cumulativeGasUsed= bytesToUint(rawReceipt.cumulativeGasUsed);
+        receipt.gasUsed = bytesToUint(rawReceipt.gasUsed);
+        receipt.status = bytesToUint(rawReceipt.status);
+        receipt.transactionIndex = bytesToUint(rawReceipt.transactionIndex);
+        receipt.blockNumber = bytesToUint(rawReceipt.blockNumber);
+        receipt.logs = rawToConvertedReceiptLogs(rawReceipt.logs);
+        receipt.logsBloom = rawReceipt.logsBloom;
+        receipt.transactionHash = rawReceipt.transactionHash;
+        return receipt;
+    }
+
+    function rawToConvertedReceiptLogs(RawReceiptLog[] memory rawLogs)
+        internal
+        returns (ReceiptLog[] memory)
+    {
+        ReceiptLog[] memory logs = new ReceiptLog[](rawLogs.length);
+        for(uint i; i<rawLogs.length;i++){
+            logs[i].logAddress = rawLogs[i].logAddress;
+            logs[i].blockHash = rawLogs[i].blockHash;
+            logs[i].blockNumber = bytesToUint(rawLogs[i].blockNumber);
+            logs[i].data = rawLogs[i].data;
+            logs[i].logIndex = bytesToUint(rawLogs[i].logIndex);
+            logs[i].topics = rawLogs[i].topics;
+            logs[i].transactionIndex = bytesToUint(rawLogs[i].transactionIndex);
+            logs[i].transactionLogIndex = bytesToUint(rawLogs[i].transactionLogIndex);
+            logs[i].removed = rawLogs[i].removed;
+        }
+        return logs;
+
+    }
+
+    function bytesToUint(bytes memory b) internal pure returns (uint256){
```

Comment:
This function is used a lot but we don't have any tests for it. Alternatively, you can use assembly to convert to bytes32 and then just cast to uint